src/Tizen.Account.FidoClient/Tizen.Account.FidoClient/UafClient.cs

```diff
@@ -152,65 +159,72 @@ public static async Task<bool> CheckPolicyAsync(UafMessage uafMessage)
         /// </example>
         public static async Task<UafResponse> ProcessRequestAsync(UafMessage uafMessage, string channelBindng)
         {
+            IntPtr id = IntPtr.Zero;
+            lock (_taskCompletionMap)
+            {
+                id = (IntPtr)_responseCompletionId++;
+            }
+
+            TaskCompletionSource<UafResponse> tcsUafResponse = new TaskCompletionSource<UafResponse>();
+            _taskCompletionMap[id] = tcsUafResponse;
+            int ret = Interop.UafClient.FidoUafGetResponseMessage(uafMessage.Operation, channelBindng, _UafResponseMessageCallback, id);
+
             if (uafMessage == null)
             {
                 Log.Error(ErrorFactory.LogTag, "Invalid request or request is null");
                 throw ErrorFactory.GetException((int)FidoErrorCode.InvalidParameter);
             }
 
-            TaskCompletionSource<UafResponse> tcs = new TaskCompletionSource<UafResponse>();
-            Interop.UafClient.FidoUafResponseMessageCallback cb = (int errorCode, string uafResponseJson, IntPtr userData) =>
-            {
-                if (uafMessage == null)
-                {
-                    Log.Error(ErrorFactory.LogTag, "Invalid request or request is null");
-                    tcs.SetException(ErrorFactory.GetException((int)FidoErrorCode.InvalidParameter));
-                }
-
-                if (errorCode != (int)FidoErrorCode.None)
-                {
-                    Log.Error(ErrorFactory.LogTag, "Interop callback failed with error code: [" + errorCode + "]");
-                    tcs.SetException(ErrorFactory.GetException(errorCode));
-                }
-
-                tcs.SetResult(new UafResponse() { Response = uafResponseJson });
-            };
-
-            int ret = Interop.UafClient.FidoUafGetResponseMessage(uafMessage.Operation, channelBindng, cb, IntPtr.Zero);
             if (ret != (int)FidoErrorCode.None)
             {
                 Log.Error(ErrorFactory.LogTag, "Interop API failed with error code: [" + ret + "]");
                 throw ErrorFactory.GetException(ret);
             }
 
-            return await tcs.Task;
+            return await tcsUafResponse.Task.ConfigureAwait(true);
```

Comment:
```suggestion
            return await tcsUafResponse.Task.ConfigureAwait(false);
```
because no more code to execute on mainthread.